### PR TITLE
Eliminable differentiated states

### PR DESCRIPTION
**Pull request by jbaayen**
This is, again, useful for RTC-Tools Channel Flow, so that we can eliminate differentiated states from optimization problems if desired.

### CODE REVIEW

**Comment by jackvreeken** [comment]
You probably already figured this out, but the tests are failing on the f-format string (not supported yet in Python 3.5). You would have to use one of the older ways of formatting strings.

File: src/pymoca/backends/casadi/model.py [hunk 1]
```diff
@@ -568,21 +580,61 @@ def extract_assignment(eq):
...
+
             reduced_equations = []
             for eq in self.equations:
                 variable, value = extract_assignment(eq)
                 if variable is not None:
+                    if variable.name() in states:
+                        # Mark derivative state for replacement too.
+                        derivative = get_derivative(value)
+
+                        state_index = list(states.keys()).index(variable.name())
```
Comment (jackvreeken) [inline src/pymoca/backends/casadi/model.py]: Using `.index` could be dangerously slow, and will likely come back to bite us. It's probably better to make a dict `MX state -> MX derivative` or `str state (name) -> MX derivative`. Note that all other lookups like `eq.dep(0).name() in states` are already O(1).

**Comment by jackvreeken** [comment]
@jbaayen Pushed one commit on top to get rid of the `.index`. Could you check if you agree with my implementation?

**Comment by jbaayen** [comment]
Useful improvement -- thanks! Looks good.

**Pull request merged**